dace/transformation/passes/simplification/control_flow_raising.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+from typing import Optional, Tuple
+import networkx as nx
+from dace import properties
+from dace.sdfg.analysis import cfg as cfg_analysis
+from dace.sdfg.sdfg import SDFG, InterstateEdge
+from dace.sdfg.state import ConditionalBlock, ControlFlowBlock, ControlFlowRegion
+from dace.sdfg.utils import dfs_conditional
+from dace.transformation import pass_pipeline as ppl, transformation
+from dace.transformation.interstate.loop_lifting import LoopLifting
+
+
+@properties.make_properties
+@transformation.experimental_cfg_block_compatible
+class ControlFlowRaising(ppl.Pass):
+
+    CATEGORY: str = 'Simplification'
+
+    def modifies(self) -> ppl.Modifies:
+        return ppl.Modifies.CFG
+
+    def should_reapply(self, modified: ppl.Modifies) -> bool:
+        return modified & ppl.Modifies.CFG
+
+    def _lift_conditionals(self, sdfg: SDFG) -> int:
+        cfgs = list(sdfg.all_control_flow_regions())
+        n_cond_regions_pre = len([x for x in sdfg.all_control_flow_blocks() if isinstance(x, ConditionalBlock)])
+
+        for region in cfgs:
+            dummy_exit = region.add_state('__DACE_DUMMY')
+            for s in region.sink_nodes():
+                if s is not dummy_exit:
+                    region.add_edge(s, dummy_exit, InterstateEdge())
```

Comment:
Weird way of adding a dummy node. make a list(sink_nodes()) (or set) your iterator and always add the edge.